src/coreclr/jit/ssabuilder.cpp

```diff
@@ -1565,8 +1565,8 @@ void SsaBuilder::SetupBBRoot()
         return;
     }
 
-    BasicBlock* bbRoot = BasicBlock::bbNewBasicBlock(m_pCompiler, BBJ_NONE);
-    bbRoot->bbFlags |= BBF_INTERNAL;
+    BasicBlock* bbRoot = BasicBlock::bbNewBasicBlock(m_pCompiler, BBJ_ALWAYS, m_pCompiler->fgFirstBB);
+    bbRoot->bbFlags |= (BBF_INTERNAL | BBF_NONE_QUIRK);
```

Comment:
Seems odd that this can't just use `fgEnsureFirstBBIsScratch`. Something for another day I guess.